angular/core/components/change_volume_form/change_volume_form.coffee

```diff
@@ -0,0 +1,42 @@
+angular.module('loomioApp').factory 'ChangeVolumeForm', ->
+  templateUrl: 'generated/components/change_volume_form/change_volume_form.html'
+  controller: ($scope, model, setDefault, FormService, CurrentUser, FlashService) ->
+    $scope.model = model.clone()
+    $scope.volumeLevels = ["loud", "normal", "quiet"]
+    $scope.buh = {}
+    $scope.buh.volume = model.volume()
+    $scope.setDefault = setDefault if setDefault
+
+    $scope.translateKey =
+      if setDefault
+        "change_volume_form.set_membership_default"
+      else
+        "change_volume_form.#{$scope.model.constructor.singular}"
+
+    $scope.title =
+      if setDefault
+        ''
+      else
+        model.title or model.groupName()
+
+
+    $scope.submit = ->
+      $scope.isDisabled = true
```

Comment:
I don't follow; I just gave that above code a smoke test and it seems to work alright (at least all of the flash messages are showing up correctly.)

The idea is that `submitFn` is a function which takes the second FormService argument (in this case, $scope.model), as an argument.

(ie, we go: 
```
FormService.submit = (scope, model, options) ->
  options.submitFn(model)
```

So we want to write a function as submitFn which takes the model and does what we want, and then we get all of the magic of the form service like flash messaging, error handling, form disabling, and closing the modal automatically.